packages/link-modules/src/loaders/connection.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  InternalModuleDeclaration,
+  LoaderOptions,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+} from "@medusajs/modules-sdk"
+import { MedusaError, ModulesSdkUtils } from "@medusajs/utils"
+
+import { ModulesSdkTypes } from "@medusajs/types"
+import { asValue } from "awilix"
+import { createConnection } from "../utils"
+
+export function connectionLoader(entity) {
+  return async (
+    {
+      options,
+      container,
+    }: LoaderOptions<
+      | ModulesSdkTypes.ModuleServiceInitializeOptions
+      | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+    >,
+    moduleDeclaration?: InternalModuleDeclaration
+  ): Promise<void> => {
+    if (
+      moduleDeclaration?.scope === MODULE_SCOPE.INTERNAL &&
+      moduleDeclaration.resources === MODULE_RESOURCE_TYPE.SHARED
+    ) {
+      return
+    }
+
+    const customManager = (
+      options as ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+    )?.manager
+
+    if (!customManager) {
+      const dbData = ModulesSdkUtils.loadDatabaseConfig("link_modules", options)
+      await loadDefault({ database: dbData, container, options })
+    } else {
+      container.register({
+        manager: asValue(customManager),
+      })
```

Comment:
todo: can you add a todo here that once that [pr](https://github.com/medusajs/medusa/pull/4626) is merged we should refactor, I might extract that to an util as well since it is always the same for us and if not then we have a custom impl